bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/events/ItemEventFactory.java

```diff
@@ -97,8 +97,8 @@ private String getItemName(String topic) {
         return topicElements[2];
     }
 
-    private Object parse(String className, String valueToParse) throws Exception {
-        Class<?> stateClass = Class.forName(className);
+    private Object parse(String typeName, String valueToParse) throws Exception {
+        Class<?> stateClass = Class.forName("org.eclipse.smarthome.core.library.types." + typeName);
```

Comment:
Should we get the package name from a class like the "UnDefType" or at least extract it as a constant?